rad/src/operators/mod.rs

```diff
@@ -0,0 +1,76 @@
+// FIXME: https://github.com/rust-num/num-derive/issues/20
+#![allow(clippy::useless_attribute)]
+
+use crate::error::RadResult;
+use crate::script::RadonCall;
+use crate::types::RadonTypes;
+
+use num_derive::FromPrimitive;
+use std::fmt;
+
+pub mod mixed;
+pub mod string;
+
+#[derive(Debug, FromPrimitive, PartialEq)]
+pub enum RadonOpCodes {
+    /// Only for the sake of allowing catch-alls when matching
+    Fail = -1,
+    // Multi-type operator codes start at 0x00
+    /// Identity operator code
+    Identity = 0x00,
+    // Boolean operator codes start at 0x10
+    // Integer operator codes start at 0x20
+    // Float operator codes start at 0x30
+    // Null operator codes start at 0x40
+    // String operator codes start at 0x50
+    /// Parse Mixed from JSON string
+    ParseJson = 0x53,
+    // Array operator codes start at 0x60
+    // Map operator codes start at 0x70
+    // Mixed operator codes start at 0x80
+    // Result operator codes start at 0x90
+}
+
+impl fmt::Display for RadonOpCodes {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "RadonOpCodes::{:?}", self)
+    }
+}
+
+pub trait Operable {
+    fn operate(self, call: &RadonCall) -> RadResult<RadonTypes>;
+}
+
+pub fn operate(input: RadonTypes, call: &RadonCall) -> RadResult<RadonTypes> {
+    match input {
+        RadonTypes::String(radon_string) => radon_string.operate(call),
+        RadonTypes::Mixed(radon_mixed) => radon_mixed.operate(call),
+    }
+}
```

Comment:
If `RadonTypes` variants, all contain types that implement `Operable`, then `RadonTypes` may implement it to with this exact same implementation, if we do so then this function will just be:

```
pub fn operate(input: RadonTypes, call: &RadonCall) -> RadResult<RadonTypes> {
    input.operate(call)
}
```